core/parse-xml.c

```diff
@@ -448,6 +448,24 @@ static void get_dc_type(char *buffer, enum dive_comp_type *dct)
 	}
 }
 
+/* For divemode_text[] (defined in dive.h) determine the index of
+ * the string contained in the xml divemode attribute and passed
+ * in buffer, below. Typical xml input would be:
+ * <event name='modechange' divemode='OC' /> */
+static void event_divemode(char *buffer, int *value)
+{
+	int size = trimspace(buffer);
+	if (size >= MAX_EVENT_NAME)
+		size = MAX_EVENT_NAME-1;
+	buffer[size] = 0x0;
+	for (int i=0; i<4; i++) {
```

Comment:
whitespace. Also, hardcoding 4 here is definitely not good code

Reply:
please, modify to `i < 4` but also replace `4` with a define like @dirkhh suggests.